core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -109,6 +109,8 @@ trait MetadataCache {
   def getRandomAliveBrokerId: Option[Int]
 
   def features(): Features
+
+  def isInitialized(): Boolean
```

Comment:
Since KRaft brokers have the concept of fencing and metadata write requests are forwarded to the controller, this isn't really necessary. Can we make this method ZkMetadataCache-only?

Reply:
Sure, just updated this with my latest patch.